### PR TITLE
Add deployment metadata.generation metric

### LINKED ISSUE #32: Missing deployment metric metadata.generation
I'm trying to use the metrics provided by this project but it's missing the metadata.generation metric to check for status of a deployment. It would be great if that could be added in!

Below is from the kubernetes docs about deployments. I'm using that as a guideline to check if deployments fail or not and then send out an alert if it does fail.
> 
After creating or updating a Deployment, you would want to confirm whether it succeeded or not. The simplest way to do this is through kubectl rollout status.

> 
This verifies the Deployment’s **.status.observedGeneration >= .metadata.generation**, and its up-to-date replicas (**.status.updatedReplicas**) matches the desired replicas (**.spec.replicas**) to determine if the rollout succeeded. If the rollout is still in progress, it watches for Deployment status changes and prints related messages.
Note that it’s impossible to know whether a Deployment will ever succeed, so if the above command doesn’t return success, you’ll need to timeout and give up at some point.
Additionally, if you set **.spec.minReadySeconds**, you would also want to check if the available replicas (**.status.availableReplicas**) matches the desired replicas too.

### PR DESCRIPTION
**Pull request by shamil**
This PR adds deployment `metadata.generation` metric
Fixes #32

### CODE REVIEW

**Comment by shamil** [comment]
I signed it!

**Comment by brancz** [comment]
Looks pretty good. Could you also add documentation on the `README.md`?

And to accept the PR we also need you to sign the cla/linuxfoundation.

File: deployment.go [hunk 1]
```diff
@@ -61,6 +61,12 @@ var (
 		"Whether the deployment is paused and will not be processed by the deployment controller.",
 		[]string{"namespace", "deployment"}, nil,
 	)
+
+	descDeploymentMetadataGeneration = prometheus.NewDesc(
+		"kube_deployment_metadata_generation",
+		"Sequence number representing a specific generation of the desired state",
```
Comment (brancz) [inline deployment.go]: Missing trailing colon.

**Comment by shamil** [comment]
Thanks for review @brancz 
- `cla/linuxfoundation`  signed.
- updated `README.md` and added missing trailing colon

**Comment by brancz** [comment]
Hmm .. looks like something with the CLA is not correct. Can you please make sure you confirmed your email and everything, and that your commits are created with that same email you used to sign the CLA?

**Comment by shamil** [comment]
OK, squashed the commits, looks good now...

**Comment by brancz** [comment]
Alright! LGTM 👍 !

**Comment by brancz** [comment]
As mentioned in #37 we'll give the pending PRs some time until probably early next week before we do another release. If it's super urgent for you to have these metrics, then I would suggest to build from master until the next release. Thanks for the effort! 🎉

**Pull request merged**